### ISSUE TITLE
Performance issues

### ISSUE THREAD

**Issue by nchapman** [mozilla/activity-stream#934]
Hi all,

Let's take a look at some of the performance issues mentioned here:

https://medium.com/@manojhatax/hi-nick-e961b6f44f35#.kzu0tvn1k

> 1. The new tab page extension to include my history and timeline was a good start. I stopped using History completely. However, items I removed from the timeline seemed to not persist across tabs.
2. Adding the Experiment to my browser slowed it down noticeably. The URL bar autocomplete, in particular, was impacted to the point of becoming unusable unless I waited a few seconds after creating a new tab to start typing a URL.
3. Memory usage of Firefox on my MacBook Air increased by about 15%. This observation is based on random sampling using Activity Monitor, but is a concern nonetheless.
4. The Front/Back buttons stopped working correctly once a browsing session had been active for more than 8-hours. Firefox has become stable enough for me to not quit the browser instance unless I absolutely have to (typically for security updates).

> The net-net for me is that the current prototype is too immature for me to use every day. I look forward to the next release, and I hope that stability and performance issues are identified, prioritized and fixed in parallel with new feature development in the future.

**Comment by jaredkerim** [comment]
Closing as a duplicate of #868
